[Android Editor] Error parsing `editor_settings-4.3.tres`

### Tested versions

- Reproducible in Godot Android Editor 4.3 beta 2
- Not reproducible in Godot Android Editor 4.2.2 stable

### System information

Android 12, Android 14

### Issue description

There is a regression that causes the `editor_settings-4.3.tres` file to be randomly truncated. This seems to be a race condition as the issue doesn't always occur, but when it does, it first causes the editor to crash when re-opening the project, then it causes the editor to crash every time it's opened with the following error logs:

```
E  USER ERROR: /data/data/org.godotengine.editor.v4.debug/files/config/godot/editor_settings-4.3.tres:1 - Parse Error: 
E     at: open (scene/resources/resource_format_text.cpp:1056)
E  USER ERROR: Failed loading resource: /data/data/org.godotengine.editor.v4.debug/files/config/godot/editor_settings-4.3.tres. Make sure resources have been imported by opening the project in the editor at least once.
E     at: _load (core/io/resource_loader.cpp:282)
A  Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0 in tid 8173 (GLThread 30), pid 8145 (editor.v4.debug)
---------------------------- PROCESS STARTED (8196) for package org.godotengine.editor.v4.debug ----------------------------
A  Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0 in tid 8173 (GLThread 30), pid 8145 (editor.v4.debug)
```

The `editor_settings-4.3.tres` file ends up empty.


### Steps to reproduce

- Download Godot Android Editor 4.3 beta 2
- Load a sample project from the asset library. E.g: https://godotengine.org/asset-library/asset/2174
- Open the project in the editor
- After the project has loaded, select `Project` -> `Quit to project list` to return to the project list
- In the project list, press the back button to exit the editor
- Re-open the editor
- Re-launch the sample project

Note: As mentioned in the description, this seems to be a race condition so it may take several attempts before it repros, and even then it's not guaranteed to repro

### Minimal reproduction project (MRP)

https://godotengine.org/asset-library/asset/2174

--- a/editor/editor_settings.cpp
+++ b/editor/editor_settings.cpp
@@ -1063,13 +1063,13 @@ void EditorSettings::create() {
 		}
 
 		singleton = ResourceLoader::load(config_file_path, "EditorSettings");
-		singleton->set_path(get_newest_settings_path()); // Settings can be loaded from older version file, so make sure it's newest.
-
 		if (singleton.is_null()) {
 			ERR_PRINT("Could not load editor settings from path: " + config_file_path);
+			config_file_path = get_newest_settings_path();
 			goto fail;
 		}
 
+		singleton->set_path(get_newest_settings_path()); // Settings can be loaded from older version file, so make sure it's newest.
 		singleton->save_changed_setting = true;
 
 		print_verbose("EditorSettings: Load OK!");

